sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/backup_client.hpp

```diff
@@ -21,9 +21,20 @@
 #include <string>
 #include <thread>
 
-using namespace Azure::Security::KeyVault::Administration::Models;
-
 namespace Azure { namespace Security { namespace KeyVault { namespace Administration {
+  /**
+   * @brief Define the options to create an SDK Keys client.
+   *
+   */
+  struct BackupClientOptions final : public Azure::Core::_internal::ClientOptions
+  {
+    /**
+     * @brief Service Version used.
+     *
+     */
+    const std::string ApiVersion{"7.5"};

```

Comment:
Const members are problematic, I recommend to avoid them, if possible. For example, you can't reassign options anymore.

Our guidelines are also against it: https://github.com/Azure/azure-sdk/blob/5b101623dd57f062c5d5e13ddc0aac3b5b4ed4b2/docs/cpp/implementation.md#const-and-reference-members ("Must not declare types with const or reference members. ")

@gearama @LarryOsterman

Reply:
That's a good catch. More importantly, this is a field that is supposed to be customizable by the customer, so it *shouldn't* be const (ignoring the guidelines issue).

Reply:
I think we should search our SDK for more occurrences of this, and fix them all if there are more, and also have ApiView signal if it sees const members.

Reply:
> That's a good catch. More importantly, this is a field that is supposed to be customizable by the customer, so it _shouldn't_ be const (ignoring the guidelines issue).

Larry we should flag this in APIView (seems APIView is becoming our linter (lol))

Reply:
I think that's a reasonable request. 

Another one on similar lines is any struct/class whose name ends with `ClientOptions` being in the "Models" namespace (types in the Models namespace are supposed to be types transferred to or from the service).